mlrun/projects/pipelines.py

```diff
@@ -591,6 +591,22 @@ def run(
             artifact_path=artifact_path,
             cleanup_ttl=workflow_spec.cleanup_ttl,
         )
+
+        # The user provided workflow code might have made changes to function specs that require cleanup
+        for func in project.spec._function_objects.values():
+            try:
+                func.spec.rollback_fields()
+            except AttributeError:
+                logger.debug(
+                    f"Function of type {type(func)} doesn't require a spec field rollback"
+                )
```

Comment:
our logger knows how to handle kwargs so once we change human readable to json, it structure the key-val in an organized way
```suggestion
                logger.debug(
                    "Function does not require field rollback", func_type=type(func),
                )
```

this apply for the rest of the added log lines

Reply:
Good point. I improved it for us.